lightning/src/ln/features.rs

```diff
@@ -152,9 +152,12 @@ impl NodeFeatures {
 	/// relevant in a node-context features and creates a node-context features from them.
 	pub(crate) fn with_known_relevant_init_flags(init_ctx: &InitFeatures) -> Self {
 		let mut flags = Vec::new();
-		if init_ctx.flags.len() > 0 {
-			// Pull out data_loss_protect and upfront_shutdown_script (bits 0, 1, 4, and 5)
-			flags.push(init_ctx.flags.last().unwrap() & 0b00110011);
+		for (i, feature_byte)in init_ctx.flags.iter().enumerate() {
```

Comment:
You probably want just init_ctx.flags.len() > 0 { init_ctx.flags[0] } instead of adding a loop just to get the first element.

Reply:
Don't we want to add all elements? Otherwise we'll keep sending legacy onion payloads

Reply:
Hmm, let me know how the docs can be improved here but I think there's a miscommunication about what this fn should do -

the thinking is, the flags which a node sets in their init message are definitely the most up-to-date copy we have of what the node supports. Thus, in places where we want to hold a NodeFeatures (just from a Rust types PoV) we need to copy over the bits that we are gonna want to check for. For unknown bits, there is both no reason to copy them over (we're never gonna check them) and we also aren't sure they are allowed to exist in other contexts. Thus, we don't need to copy all the bits that are there, only the ones we know how to interpret and know are both I and N contexts.

Reply:
Ah, understood. I think this miscommunication was on me, but I added one clarifying comment. Let me know if I'm not blanking out the correct features now.